compiler/rustc_lint/messages.ftl

```diff
@@ -5,6 +5,10 @@ lint_array_into_iter =
     .use_explicit_into_iter_suggestion =
         or use `IntoIterator::into_iter(..)` instead of `.into_iter()` to explicitly iterate by value
 
+lint_async_fn_in_trait = usage of `async fn` in trait is discouraged because they do not automatically have auto trait bounds
+    .note = you can suppress this lint if you plan to use the trait locally, for concrete types, or do not care about auto traits like `Send` on the future
```

Comment:
Can we perhaps be more helpful to new users?  Like so:

```
you can suppress this lint if you do not care about auto traits like `Send` on the future with the `#[allow(async_fn_in_trait)]` attribute
```

Reply:
I'd rather just make it a suggestion, but also, I don't think any other lint goes out of its way to tell you how to suppress it.

Reply:
I'm sorry, but what's a structured suggestion?

Also, yes, other lints don't do this, but most other lints aren't elevated to be shown by default unless it's guaranteed to not have false positives. Additionally, new-comers probably wouldn't understand a lot about what `Send` bounds are and why it's needed. While yes, they should eventually learn it, it only increases the barrier to getting started with rust for them. Giving them a simple suggestions is more of a "here's the simple, ugly way out, but you should probably look into this when you understand what's happening"

Reply:
On second thought, it would also encourage users to do the wrong thing unless they know what they're doing, which isn't exactly ideal.

Hmm, idk. I'm conflicted now. I've put both sides of the coin on the table. Open for suggestions